src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/EventCounter/EventCounterTrigger.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.EventCounter
+{
+    /// <summary>
+    /// Trigger that detects when the specified event source counter value is held
+    /// above, below, or between threshold values for a specified duration of time.
+    /// </summary>
+    internal sealed class EventCounterTrigger :
+        ITraceEventTrigger
+    {
+        private readonly CounterFilter _filter;
+        private readonly EventCounterTriggerImpl _impl;
+        private readonly string _providerName;
+
+        public EventCounterTrigger(EventCounterTriggerSettings settings)
+        {
+            if (null == settings)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            Validate(settings);
+
+            _filter = new CounterFilter(settings.CounterIntervalSeconds);
+            _filter.AddFilter(settings.ProviderName, new string[] { settings.CounterName });
+            
+            _impl = new(settings);
+
+            _providerName = settings.ProviderName;
+        }
+
+        public IDictionary<string, IEnumerable<string>> GetProviderEventMap()
+        {
+            return new Dictionary<string, IEnumerable<string>>()
```

Comment:
iirc provider names are case insensitive names while counter names are case sensitive. I'm not sure that's the case for all trace events though, but certainly for counters you may need to modify the dictionary to account for this.

Reply:
It does need to be created each time because the event provider name is configurable per instance of the trigger. However, I can make the types the readonly versions and cache the event name list.

Reply:
Oh, I think I now better understand the question. You're asking if we can cache the dictionary per event provider. I can do that.